packages/yourloops/components/dialogs/join-team/confirm-code-team.tsx

```diff
@@ -0,0 +1,113 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { useTranslation } from 'react-i18next'
+import React, { createRef, useEffect } from 'react'
+import { type Team } from '../../../lib/team'
+import Box from '@mui/material/Box'
+import DialogTitle from '@mui/material/DialogTitle'
+import DialogContent from '@mui/material/DialogContent'
+import DialogContentText from '@mui/material/DialogContentText'
+import TextField from '@mui/material/TextField'
+import DialogActions from '@mui/material/DialogActions'
+import Button from '@mui/material/Button'
+import useConfirmCodeTeam from './confirm-code-team.hooks'
+
+export interface ConfirmTeamProps {
+  onClickCancel: () => void
+  onCompleteStep: (team: Team, teamId?: string) => void
+  teamName: string
+}
+
+export const ConfirmCodeTeam = (props: ConfirmTeamProps): JSX.Element => {
+  const { onClickCancel, onCompleteStep, teamName } = props
+  const { t } = useTranslation()
+  const inputRef = createRef<HTMLInputElement>()
+  const { handleChangeCode, handleClickJoinTeam, joinButtonDisabled, idCode } = useConfirmCodeTeam({ onCompleteStep })
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus()
+    }
+  }, [inputRef])
+
+  return (
+    <React.Fragment>
+      <Box textAlign="center">
+        <DialogTitle>
+          <strong id="team-add-dialog-title" data-testId="team-add-dialog-title">
+            {teamName
+              ? t('modal-add-medical-specific-team', { careteam: teamName })
+              : t('modal-add-medical-team')
+            }
+          </strong>
+        </DialogTitle>
+
+        <DialogContent>
+          <Box display="flex" flexDirection="column" alignItems="center">
+            <DialogContentText data-testId="label-dialog">
+              {teamName
+                ? (t('modal-add-medical-team-code'))
+                : (t('modal-add-medical-team-code-no-invite'))
+              }
+            </DialogContentText>
+            <Box>
+              <TextField
+                id="team-add-dialog-field-code"
+                variant="standard"
+                value={idCode}
+                onChange={handleChangeCode}
+                fullWidth
+                inputRef={inputRef}
+                style={{ width: '55%' }}
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏ ugly in code but nice on screen 😜‏
```suggestion
                InputProps={{ sx: { width: 120, '& > input': { textAlign: 'center' } } }}
```